modules/ROOT/pages/cryptography-troubleshooting.adoc

```diff
@@ -0,0 +1,96 @@
+* If you’re running the app using Mule from the command line, the app log is visible in your OS console.
+
+Unless the log file path is customized in the app’s log file `log4j2.xml`, view the app log in the default location `MULE_HOME/logs/<app-name>.log`.
+
+== Enable Cryptography Module Debug Logging
+
+To begin troubleshooting Cryptography Module, enable debug logging to see the exact error messages:
+
+. Access Anypoint Studio and navigate to the *Package Explorer* view.
+. Open your application’s project name.
+. Open the `src/main/resources` path folder.
+. Open the `log4j2.xml` file inside the folder.
+. Add the following line:
++
+`<AsyncLogger name=”com.mulesoft.modules.cryptography" level="DEBUG" />`
+
+[start=6]
+. Save your changes.
+. Click the project name in *Package Explorer* and then click *Run* > *Run As* > *Mule Application*.
+
+== Troubleshoot PGP Encryption and Decryption
+
+During PGP encryption, the sender of the message must encrypt its content using the receiver’s _public key_. So, whenever you want to encrypt messages in your Mule app using someone else’s public key, in the *Crypto Pgp* global configuration add the _receiver_ public keyring file in the *Public keyring* field.
+
+During PGP decryption, the receiver of the message must use its _private key_ to decrypt the contents of a message that was encrypted using a public key. To decrypt the message, in the *Crypto Pgp* global configuration add the _receiver_ private keyring file in the *Private keyring* field.
+
+.Crypto Pgp Global configuration with Public keyring and Private keyring fields
+image::mruntime-crypto-pgp-global-config.png[Crypto Pgp Global configuration with Public keyring and Private keyring fields]
+
+== Troubleshoot PGP Signature
+
+In addition to encrypting, you can sign a message, the signature provides an integrity check of the original message.
+
+To create a signature, in the *Crypto Pgp* global configuration add the _sender_ private keyring file in the *Private keyring* field.
+To validate a signature, in the *Crypto Pgp* global configuration add the _sender_ public keyring file in the *Public keyring* field.
+
+=== PGP Signature Types and Operations
+
+Cryptography module has two PGP signature operations:
+
+*Pgp sign* +
+Creates a PGP armored signature in ASCII format. The signing is done with the private key of the sender, so the secret passphrase must be provided.
+*Pgp sign binary* +
+Creates a PGP binary signature. The signing is done with the private key of the sender, so the secret passphrase must be provided.
+
+There are two types of signatures:
+
+* Attached signature +
+Given a signed document, you can either check the signature or check the signature and recover the original document.
+
+* Detached signature +
+The signature and the document are created separately.
+
+Currently, Cryptography module supports only validation of detached signatures.
```

Comment:
```suggestion
Currently, the Cryptography module supports validation of detached signatures only.
```
Also, if this is true, than why do we define attached signatures in terms of actions that the user takes?